SENDING EMAILS BACKEND

melcidar opened the pull request:
Sending emails backend

--------------------------------------------------------------------
 MAIN
 * sending mails using Nodemailer
--------------------------------------------------------------------
 OTHER
 * /////////
--------------------------------------------------------------------

becarselman committed:
Initial commit

ElvaRivle committed:
Initial folder structure (#2)

* npm init and express added

* entry point js file

* important folders added

* express hello world example

* nodemon and script `dev` added

installed nodemon locally as development dependency

so we must use npm run to use it

ElvaRivle committed:
FEATURE/ElvaRivle/DfWCIPVU

--------------------------------------------------------------------
 MAIN
 * successfully connected to MongoDB
 * created User model
 * seeded default users to DB
--------------------------------------------------------------------
 OTHER
 * added all necessary packages
 * loading connection string from .env file
 * listening and logging mongoose events
--------------------------------------------------------------------

ElvaRivle committed:
FEATURE/ElvaRivle/DfWCIPVU

--------------------------------------------------------------------
 MAIN
 * seeder converted to script
 * README added
--------------------------------------------------------------------
 OTHER
 * changed default users for seeding
--------------------------------------------------------------------

ElvaRivle committed:
FEATURE/ElvaRivle/DfWCIPVU (#4)

* FEATURE/ElvaRivle/DfWCIPVU

--------------------------------------------------------------------
 MAIN
 * successfully connected to MongoDB
 * created User model
 * seeded default users to DB
--------------------------------------------------------------------
 OTHER
 * added all necessary packages
 * loading connection string from .env file
 * listening and logging mongoose events
--------------------------------------------------------------------

* FEATURE/ElvaRivle/DfWCIPVU

--------------------------------------------------------------------
 MAIN
 * seeder converted to script
 * README added
--------------------------------------------------------------------
 OTHER
 * changed default users for seeding
--------------------------------------------------------------------

ElvaRivle committed:
FEATURE/ElvaRivle/DfWCIPVU-v2

--------------------------------------------------------------------
 MAIN
 * implemented login (on successful login JWT is returned)
 * added login service, controller and routes
--------------------------------------------------------------------
 OTHER
 * README typo
 * new JWT_SECRET env variable
 * closed mongoose connection on seed failure
 * globally enabled JSON body parsing
--------------------------------------------------------------------

ElvaRivle committed:
Merge branch 'develop' into FEATURE/ElvaRivle/DfWCIPVU-v2

ElvaRivle committed:
FEATURE/ElvaRivle/DfWCIPVU-v2

--------------------------------------------------------------------
 MAIN
 * all errors added to errors.js file
 * all env variables added to env.js file
 * added user repository
--------------------------------------------------------------------
 OTHER
 * rewrote login service and controller
 * controller made minimalistic, all logic moved to service
--------------------------------------------------------------------

ElvaRivle committed:
FEATURE/ElvaRivle/DfWCIPVU-v2

--------------------------------------------------------------------
 MAIN
 * renamed `utils` folder to `configuration`
 * added EMAIL_INVALID_FORMAT error message
--------------------------------------------------------------------
 OTHER
 * explicitly returned 200 on successful login
 * fixed misc warnings about unused requires
--------------------------------------------------------------------

ElvaRivle committed:
FEATURE/ElvaRivle/DfWCIPVU-v2

--------------------------------------------------------------------
 MAIN
 * when `nodemon` restarts, release previous connection back to pool
--------------------------------------------------------------------
 OTHER
 * capture SIGTERM as well
 * extracted app termination login in separate function

AntMarkoLovric committed:
Merge pull request #5 from becarselman/FEATURE/ElvaRivle/DfWCIPVU-v2

FEATURE/ElvaRivle/DfWCIPVU-v2

melcidar committed:
SENDING EMAILS BACKEND

--------------------------------------------------------------------
 MAIN
 * using Nodemailer
--------------------------------------------------------------------
 OTHER
 * ///////
--------------------------------------------------------------------

becarselman commented on controllers/sendemail.js:
```diff
@@ -0,0 +1,13 @@
+
+    try {
+        await sendMailService.sendEmail(email)
+        res.send('Email Sent!')
```
please use res.json, and also send status OK

becarselman commented on controllers/sendemail.js:
```diff
@@ -0,0 +1,13 @@
+        await sendMailService.sendEmail(email)
+        res.send('Email Sent!')
+    } catch (error) {
+        console.error(error)
```
remove console log.

becarselman commented on services/sendemail.js:
```diff
@@ -0,0 +1,30 @@
+const nodemailer = require("nodemailer");
+
+exports.sendEmail = async (email) => {
```
this is good, But we should be able to use this for all types of emails. Make it so that it reusable. You can create new folder with email templates and then If i call this service function from for example login service I just pass email template and everything works fine

becarselman commented on services/sendemail.js:
```diff
@@ -0,0 +1,30 @@
+exports.sendEmail = async (email) => {
+    // create reusable transporter object using the default SMTP transport
+    let transporter = nodemailer.createTransport({
+        host: "smtp.ethereal.email",
```
all of this information need to be moved into env

melcidar committed:
SENDING EMAILS BACKEND

--------------------------------------------------------------------
 MAIN
 * changed controllers/sendemail.js, used res.json and added status OK
 * removed console log
 * moved information from services/sendemail.js to new .env.sendemail file
--------------------------------------------------------------------
 OTHER
 * making function sendEmail reusable and template is yet to be added
--------------------------------------------------------------------

melcidar committed:
SENDING EMAILS BACKEND

--------------------------------------------------------------------
 MAIN
 * added template and changed sendEmail function
--------------------------------------------------------------------
 OTHER
 * could not test it because of issues with starting localhost
--------------------------------------------------------------------

becarselman commented on .env.sendemail:
```diff
@@ -0,0 +1,5 @@
+SMTP_HOST=smtp.office365.com
```
Please use one env already added. @ElvaRivle  can help you with this. and Remember this Never push env to github.

becarselman commented on controllers/sendemail.js:
```diff
@@ -0,0 +1,12 @@
+        await sendMailService.sendEmail(email)
+        res.json({message: 'Email Sent!', status: 'OK'})
+    } catch (error) {
+        res.status(500).send('Failed to send email')
```
we already have a file for error message in configuration/error.js You can use it

becarselman commented on controllers/sendemail.js:
```diff
@@ -0,0 +1,12 @@
+
+    try {
+        await sendMailService.sendEmail(email)
+        res.json({message: 'Email Sent!', status: 'OK'})
```
When I say you miss status I don't mean this. here is example of what status I mean https://www.geeksforgeeks.org/express-js-res-status-function/

becarselman commented on services/sendemail.js:
```diff
@@ -0,0 +1,36 @@
+    // create reusable transporter object using the default SMTP transport
+    let transporter = nodemailer.createTransport({
+      host: process.env.SMTP_HOST,
+      port: process.env.SMTP_PORT,
```
move this process.env in configuration/env.js file

becarselman commented on services/sendemail.js:
```diff
@@ -0,0 +1,36 @@
+    const info = await transporter.sendMail(msg);
+
+    console.log("Message sent: %s", info.messageId);
+    // Message sent: <b658f8ca-6296-ccf4-8306-87d57a0b4321@example.com>
```
remove this comment

melcidar committed:
SENDING EMAILS BACKEND

--------------------------------------------------------------------
 MAIN
 * added new error message in error file
 * added res.json and added status in controllers/sendemail.js
 * moved process.env to configuration/env.js file
 * removed comment from services/sendemail.js
--------------------------------------------------------------------
 OTHER
 * i will delete .env.sendemails and check how to use existing env file
--------------------------------------------------------------------

melcidar committed:
SENDING EMAIL BACKEND

--------------------------------------------------------------------
 MAIN
 * fixing errors in controllers/sendemail.js
--------------------------------------------------------------------
 OTHER
 * ////
--------------------------------------------------------------------
